src/ImageSharp/Formats/Jpeg/PdfJsPort/PdfJsJpegDecoderCore.cs

```diff
@@ -345,6 +345,11 @@ public void ParseStream(Stream stream, bool metadataOnly = false)
                 // Read on.
                 fileMarker = FindNextFileMarker(this.markerBuffer, this.InputStream);
             }
+
+            if (this.MetaData.IccProfile?.CheckIsValid() == false)
```

Comment:
Can we move this to `AssignResolution` and rename that to `InitDerivedMetaDataProperties`, The `return` above prevents us reading more than we need to when we are identifying images.

Reply:
I initially wanted to put it there, but it's only called from Decode and Identify and not in ParseStream. In the go port it's called at the end of the ParseStream method (and by extension in Decode and Identify).
Should we move `AssignResolution` there as well? I changed the `return` you mentioned to a `break` so things after the `while` loop will be executed.

Reply:
Both `Decode` and `Identify` call `ParseStream` in both decoders. We want to return at that point in PdfJs decoder to avoid reading the entire SOS segment when we don’t need to. So using break slows us down. 

I can move the method if you like? Save you the bother.

Reply:
yes they do, but `ParseStream` is public so it can be called directly, that's the only reason I wanted to do it there.
Sorry about the return thing, I thought break behaves differently at that point. We could use a 
`goto` to jump out of the `while` (it's evil, I know :P)

and no worries, it's a small change, I can do it.

Reply:
👍 It's only public as we use it for tests. The class is internal so not available to consumers.

Reply:
ok great, then there's no reason to change anything in that regard. Will push changes in a couple minutes.